crates/router/src/connector/nexinets.rs

```diff
@@ -501,8 +527,19 @@ impl ConnectorIntegration<api::Void, types::PaymentsCancelData, types::PaymentsR
         req: &types::PaymentsCancelRouterData,
         _connectors: &settings::Connectors,
     ) -> CustomResult<RequestContent, errors::ConnectorError> {
-        let connector_req = nexinets::NexinetsCaptureOrVoidRequest::try_from(req)?;
-        Ok(RequestContent::Json(Box::new(connector_req)))
+        if let (Some(minor_amount), Some(currency)) =
+            (req.request.minor_amount, req.request.currency)
+        {
+            let amount_to_cancel =
+                connector_utils::convert_amount(self.amount_converter, minor_amount, currency)?;
+            let connector_router_data = NexinetsRouterData::try_from((amount_to_cancel, req))?;
+            let connector_req =
+                nexinets::NexinetsCaptureOrVoidRequest::try_from(&connector_router_data)?;
+            Ok(RequestContent::Json(Box::new(connector_req)))
+        } else {
+            // FIXME: error to be thrown here
+            todo!("Should an error be thrown ?");
```

Comment:
Should a  [NotSupported](https://github.com/juspay/hyperswitch/blob/602f50b939f320ea9d85dff28dfe3f5c65afeb70/crates/hyperswitch_interfaces/src/errors.rs#L51) ConnectorError be thrown in this scenario as [nexinets api](https://developer.nexigroup.com/payengine/en-EU/api/payengine-api-v1/#orders-orderid-transactions-transactionid-cancel-post-parameters) requires the amount and currency